src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_sdfg.py

```diff
@@ -317,47 +321,34 @@ def visit_FunCall(
         node: gtir.FunCall,
         sdfg: dace.SDFG,
         head_state: dace.SDFGState,
-        let_symbols: dict[str, gtir_builtin_translators.LetSymbol],
+        reduce_identity: Optional[gtir_to_tasklet.SymbolExpr],
     ) -> list[gtir_builtin_translators.TemporaryData]:
         # use specialized dataflow builder classes for each builtin function
         if cpm.is_call_to(node, "cond"):
             return gtir_builtin_translators.translate_cond(
-                node, sdfg, head_state, self, let_symbols
+                node, sdfg, head_state, self, reduce_identity
             )
         elif cpm.is_call_to(node.fun, "as_fieldop"):
             return gtir_builtin_translators.translate_as_field_op(
-                node, sdfg, head_state, self, let_symbols
+                node, sdfg, head_state, self, reduce_identity
             )
         elif isinstance(node.fun, gtir.Lambda):
-            # We use a separate state to ensure that the lambda arguments are evaluated
-            # before the computation starts. This is required in case the let-symbols
-            # are used in conditional branch execution, which happens in different states.
-            lambda_state = sdfg.add_state_before(head_state, f"{head_state.label}_symbols")
```

Comment:
This is the most important change of the PR. Previous design was to allocate the results of let-lambdas as global arrays (by global I mean with lifecycle equal to the sdfg scope). The arrays were however initialized in the symbols state. In this PR, the arrays for let-lambdas results only exist in the scope of the let-lambda, which corresponds to the nested SDFG. Besides, within the nested SDFG, the let-lambdas arrays have the same name as the GTIR symbols for let-lambdas arguments, and we do not need to add an extra state.